### PR TITLE
Add NullCache for backend cache

### PR DESCRIPTION
**Pull request by gnumoksha**
Hello!

* Type: new feature

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/incubator/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change: Adds NullCache for backend cache.

Thanks

### CODE REVIEW

File: Library/Phalcon/Cache/Backend/NullCache.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    public function start($keyName, $lifetime = null)
+    {
+    }
+
+    public function stop($stopBuffer = true)
+    {
+    }
+
+    public function getFrontend()
+    {
```
Comment (sergeyklay) [inline Library/Phalcon/Cache/Backend/NullCache.php]: Could you please return Phalcon\Cache\Frontend\None here?

File: Library/Phalcon/Cache/Backend/NullCache.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+use Phalcon\Cache\BackendInterface;
+
+/**
+ * A cache adapter that does nothing.
+ */
+final class NullCache implements BackendInterface
+{
+    public function start($keyName, $lifetime = null)
+    {
```
Comment (sergeyklay) [inline Library/Phalcon/Cache/Backend/NullCache.php]: return  true?

**Comment by gnumoksha** [comment]
The requested changes were made.

File: Library/Phalcon/Cache/Backend/README.md [hunk 3]
```diff
@@ -96,5 +96,10 @@ echo $time;
 
 This adapter uses [windows cache extension](http://pecl.php.net/package/wincache) for PHP
 
+
+## Null
```
Comment (sergeyklay) [inline Library/Phalcon/Cache/Backend/README.md]:
```diff
- ## Null
+ ## NullCache
```

File: Library/Phalcon/Cache/Backend/README.md [hunk 4]
```diff
@@ -96,5 +96,10 @@ echo $time;
...
+
+Null adapter can be useful when you have dependency injection and wants to provider an Adapter that does nothing.
```
Comment (sergeyklay) [inline Library/Phalcon/Cache/Backend/README.md]: NullCache adapter can be useful when you have dependency injection and wants to provider an Adapter that does nothing.

**Comment by gnumoksha** [comment]
The second group of changes was made as requested.

File: Library/Phalcon/Cache/Backend/NullCache.php [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        return true;
+    }
+
+    public function stop($stopBuffer = true)
+    {
+    }
+
+    public function getFrontend()
+    {
+        return NoneFrontend();
```
Comment (sergeyklay) [inline Library/Phalcon/Cache/Backend/NullCache.php]: @gnumoksha Still missed the **`new`** keyword

**Comment by gnumoksha** [comment]
The third group of changes was made as requested. Also, I've created some tests.

**Comment by sergeyklay** [comment]
Thank you

**Pull request merged**